iguehi.akhimien

iguehi-akhimien opened the pull request:
This is what I have done so far. I pushed it to my branch, but I want you guys to review it so it can be merged to main.

iguehi-akhimien1 committed:
added my name and github username to the README.md

iguehi-akhimien1 committed:
edited the README.md

iguehi-akhimien1 committed:
added my user stories to the README.md

iguehi-akhimien1 committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	README.md

iguehi-akhimien1 committed:
added my user stories to the README.md

iguehi-akhimien1 committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	README.md

iguehi-akhimien1 committed:
I am setting up the Notes view actions: save, view, delete...

iguehi-akhimien1 committed:
I am setting up the Notes view actions: save, view, delete...

iguehi-akhimien1 committed:
Merge remote-tracking branch 'origin/iguehi.akhimien' into iguehi.akhimien

# Conflicts:
#	README.md

iguehi-akhimien1 committed:
I added the functionalities for add title, clear, upload and delete. I tried to add some style functionalities. It is not working yet.

iguehi-akhimien1 committed:
Merge branch 'main' into iguehi.akhimien

# Conflicts:
#	README.md

iguehi-akhimien1 committed:
This is many commits in one commit.
So, I made a couple of changes:
1. I resolved the conflict between the implementation of the notes view from the remote repo and mine. Then, I deleted NoteView.java.
2. I deleted some classes: DBNoteDataAccessObject.java and DataAccessException.java. I did not need them since I am saving the notes permanently to a database.
3. I added an Input Data class and an Output Data class.  I also added an InMemoryNoteDataAccessObject.java.
4. P.S. I am still implementing Clean Architecture.

iguehi-akhimien1 committed:
I am working on the Interactor now. I am fixing the current implementation so that it takes Input Data from the controller at the beginning of the use case and gives Output Data to the presenter after executing the use case.

iguehi-akhimien1 committed:
Merge branch 'main' into iguehi.akhimien

# Conflicts:
#	src/main/java/app/MainNoteApplication.java

lizakochel commented:
Coding style looks good, I would recommend adding tests to make sure the code works as it's supposed to. Since we both created NoteInputData classes, we can combine them into one; I looked over your code to see where you create new NoteInputData instances and I think we should be able to update the class without much problem. The updated NoteInputData class would have methods getContent(), getTitle(), and getSelectedLanguage().

iguehi-akhimien1 committed:
I implemented CA and deleted NoteAppBuilder.

iguehi-akhimien1 committed:
Merge branch 'main' into iguehi.akhimien

iguehi-akhimien1 committed:
I have started editing the README file as per the instructions on Quercus.
I have cleaned up the code in NotesView by refactoring.
I also adjusted the width of the program because of the notes view.

iguehi-akhimien1 committed:
Merge remote-tracking branch 'origin/iguehi.akhimien' into iguehi.akhimien

iguehi-akhimien1 committed:
I have started editing the README file as per the instructions on Quercus.
I have cleaned up the code in NotesView by refactoring.
I also adjusted the width of the program because of the notes view.

Comment (on merged code from main):
-Jenna, you cannot pass Note into any method in the Presenter. To update content of the note, you need to go view->controller->input boundary ->interactor->output boundary->presenter->view model->view.
- You have to update the content of the note in the in-memory DAO.
-Also, Presenter does not directly interact with View but with ViewModel.
The Presenter cannot call a method in view.
-Also, you cannot create an instance of Note in the View.
-Also, you cannot pass Note into a method in the Controller.

lizakochel reviewed (approved):
Coding style looks good, javadocs are there, doesn't seem to be any conflicts. I renamed by input and output data classes to include the word 'translation' so they don't get mixed up with yours.

lizakochel reviewed (commented):
- setTitle method in Note should return String but doesn't return anything
- In Note class, content and title are set as final variables so you can't change them in the setTitle and setContent methods
